src/VehicleSelector/VehicleSelector.spec.tsx

```diff
@@ -0,0 +1,137 @@
+  );
+
+  // Wait for the iframe to be attached in the DOM.
+  await page.waitForSelector('iframe[title="storybook-preview-iframe"]');
+
+  // Select a value from each Autocomplete component
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByLabel("Project Code *")
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByRole("option", { name: "911" })
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByLabel("Model Year *")
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByRole("option", { name: "2015" })
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByLabel("Vehicle Variant *")
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByRole("option", { name: "JS - 3.6 l6 - 397kW - 7MT -" })
+    .click();
+
+  // Check that the selected value is displayed in each Autocomplete component
+  await expect(
+    page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByTestId("vehicle-select")
+      .getByLabel("Project Code *")
+  ).toHaveValue("911");
+  await expect(
+    page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByTestId("vehicle-select")
+      .getByLabel("Model Year *")
+  ).toHaveValue("2015");
+  await expect(
+    page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByTestId("vehicle-select")
+      .getByLabel("Vehicle Variant *")
+  ).toHaveValue("JS - 3.6 l6 - 397kW - 7MT - R20");
+});
+
+/**
+ * Test to check that the Autocomplete components are enabled or disabled based on the selection of previous parameters.
+ */
+test("Autocomplete components are enabled or disabled based on the selection of previous parameters", async ({
+  page
+}) => {
+  // Go to the page with the VehicleSelector component
+  await page.goto(
+    "http://localhost:6006/?path=/story/selectors-vehicleselector--default"
+  );
+
+  // Wait for the iframe to be attached in the DOM.
+  await page.waitForSelector('iframe[title="storybook-preview-iframe"]');
+
+  // Check that "Model Year" is disabled
+  await expect(
+    page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByLabel("Model Year *")
+  ).toBeDisabled();
+  // Select a "Project Code"
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByLabel("Project Code *")
+    .click();
+  await page
+    .frameLocator('iframe[title="storybook-preview-iframe"]')
+    .getByRole("option", { name: "911" })
+    .click();
+
+  // Check that "Model Year" is now enabled
+  await expect(
+    page
+      .frameLocator('iframe[title="storybook-preview-iframe"]')
+      .getByLabel("Model Year *")
+  ).toBeEnabled();
+});
```

Comment:
I would also add a test, where e.g. the contents of "Project Code" are deleted, then check if all other fields' contents are also deleted, as they should be.